File: all-submissions/0199-binary-tree-right-side-view/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {

        Queue<TreeNode> queue = new LinkedList<>();
        List<Integer> answer = new ArrayList<>();
        if(root == null) return answer;
        queue.add(root);
        while(!queue.isEmpty()){
            int nodesInCurrentLevel = queue.size();
            for(int i=0;i<nodesInCurrentLevel; i++) {
                TreeNode node = queue.remove();
                if(i== nodesInCurrentLevel-1) {
                    answer.add(node.val);
                }
                if(node.left != null) {
                    queue.add(node.left);
                } 
                if (node.right != null){
                    queue.add(node.right);
                }
            }
        }
        return answer;
        
    }
}

File: all-submissions/0199-binary-tree-right-side-view/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        def bfs(root):
            if root is None:
                return 
            nonlocal res
            queue = deque([root])
            while queue:
                nodes_in_curr_level = len(queue)
                for i in range(nodes_in_curr_level):
                    node = queue.popleft()
                    if(i == nodes_in_curr_level - 1):
                        res.append(node.val)
                    if (node.left):
                        queue.append(node.left)
                    if(node.right):
                        queue.append(node.right)
        bfs(root)
        return res
